include/multipass/virtual_machine.h

```diff
@@ -76,8 +77,8 @@ class VirtualMachine : private DisabledCopyMove
     virtual void update_cpus(int num_cores) = 0;
     virtual void resize_memory(const MemorySize& new_size) = 0;
     virtual void resize_disk(const MemorySize& new_size) = 0;
-    virtual void add_vm_mount(const std::string& target_path, const VMMount& vm_mount) = 0;
-    virtual void delete_vm_mount(const std::string& target_path) = 0;
+    virtual std::unique_ptr<MountHandler> make_native_mount_handler(const SSHKeyProvider* ssh_key_provider,
+                                                                    std::string target, const VMMount& mount) = 0;
```

Comment:
Is there a special motivation to pass a string by copy? I remember you answered something about `QString` at some point, just checking you know what you're doing.

Reply:
Nope, no special reason and no relation to QString (which could be passed by value due to its copy-on-write implementation). My mistake. I will make sure to pass stuff by const reference accordingly :smile: